it/google-cloud-platform/src/main/java/com/google/cloud/teleport/it/gcp/bigtable/BigtableResourceManager.java

```diff
@@ -370,12 +401,26 @@ public synchronized ImmutableList<Row> readTable(String tableId) {
    * Deletes all created resources (instance and tables) and cleans up all Bigtable clients, making
    * the manager object unusable.
    *
+   * <p>If this Resource Manager was configured to use a static instance, the instance will not be
+   * cleaned up, but any created tables will be deleted.
+   *
    * @throws BigtableResourceManagerException if there is an error deleting the instance or tables
    *     in Bigtable.
    */
   @Override
   public synchronized void cleanupAll() {
     LOG.info("Attempting to cleanup manager.");
+
+    if (usingStaticInstance) {
+      try (BigtableTableAdminClient tableAdminClient =
+          bigtableResourceManagerClientFactory.bigtableTableAdminClient()) {
+        createdTables.forEach(tableAdminClient::deleteTable);
+        LOG.info(
+            "This manager was configured to use a static instance that will not be cleaned up.");
```

Comment:
Looks like the previous line deletes tables, no?

Reply:
It will delete tables that were created during the RM's lifetime. i.e. any new tables that were created by calling `createTable()` but not ones that were already present